EDGAR/parser.py

```diff
@@ -305,51 +306,56 @@ def get_page_number(self, page_location: dict, element: WebElement) -> int:
     width - the width of the tag
     is_annotated - 1 if the value is annotation, 0 otherwise.
     """
-    def get_annotation_features(self, webelements: list, annotations: dict, in_table: np.array, save: bool = False, out_path: str = 'sample.csv'):
-        COLUMN_NAMES = ["label_text","found_index","full_text", "anno_index", "anno_name", "anno_id",
-                        "anno_format","anno_ix_type",'annotation_unitref',"anno_decimals", "name",
-                        "anno_contextref","page_number","x","y", "height", "width","is_annotated", "in_table"]
-        page_location = self.find_page_location()
-        
+    def get_annotation_features(self, webelements: list, annotations: dict,in_table: np.array,save: bool = False, out_path: str = 'sample.csv'):
+        COLUMN_NAMES = ["anno_text","found_index","page_text", "span_text", "anno_index", "anno_name","anno_id",
+                        "anno_format","anno_ix_type",'anno_unitref',"anno_decimals",
+                        "anno_contextref","page_number","x","y", "height", "width","is_annotated","in_table"]
 
-        
-        df = pd.DataFrame(columns=COLUMN_NAMES).astype({"in_table":bool,"is_annotated":bool}) 
+        NUM_COLUMN = len(COLUMN_NAMES)
 
-        for idx, elem in enumerate(webelements):
-            
+        df = pd.DataFrame(columns=COLUMN_NAMES)
+        text_on_page = self.parse_text_by_page()
+        page_location = self.find_page_location()
+
+        #print(len(page_location),len(text_on_page))
+        number_Null = 0
+        for i, elem in enumerate(webelements):
             default_dict = {attribute: np.nan for attribute in COLUMN_NAMES}
             page_num, y = self.get_page_number(page_location, elem)
 
-            default_dict.update({"found_index": int(idx),"full_text": elem.text, "is_annotated": False,
-                                "x": elem.location["x"], "y": y, "page_number": page_num, "name": elem.get_attribute('name'),
```

Comment:
Looks largely good but what is 'name' and why is it missing below? is it placed elsewhere or renamed?

Reply:
name is a tag in ix 
i.e. 'dei:DocumentPeriodEndDate'

Reply:
It is missing in the line of code below it

Reply:
<img width="833" alt="Screen Shot 2022-11-21 at 3 12 19 PM" src="https://user-images.githubusercontent.com/66972539/203149555-923d6266-f25c-44ce-92a9-5d45fa76a4f3.png">

Reply:
okay awesome -> we can resolve this concern of the review.